web/pandas/pdeps/0015-do-not-require-pyarrow.md

```diff
@@ -0,0 +1,71 @@
+# PDEP-15: Do not require PyArrow as a required dependency (for pandas 3.0)
+
+- Created: 8 May 2024
+- Status: Under discussion
+- Discussion:  [#58623](https://github.com/pandas-dev/pandas/pull/58623)
+               [#52711](https://github.com/pandas-dev/pandas/pull/52711)
+               [#52509](https://github.com/pandas-dev/pandas/issues/52509)
+               [#54466](https://github.com/pandas-dev/pandas/issues/54466)
+- Author: [Thomas Li](https://github.com/lithomas1)
+- Revision: 1
+
+## Abstract
+
+This PDEP supersedes PDEP-10, which stipulated that PyArrow should become a required dependency
+for pandas 3.0. After reviewing feedback posted
+on the feedback issue [#54466](https://github.com/pandas-dev/pandas/issues/54466), we've
+decided against moving forward with this PDEP for pandas 3.0.
+
+The primary reasons for rejecting this PDEP are twofold:
+
+1) Requiring pyarrow as a dependency causes installation problems.
+
+   - Pyarrow does not fit or has a hard time fitting in space-constrained environments
```

Comment:
I think it's only conda that's been drastically reduced - install pyarrow and pandas in a fresh venv and it already hits 295 MB. And the wheel size on PyPI is still ~40MB, so we'd be noticeably increasing the load on PyPI by making PyArrow required

My current stance is: if the desire was to make PyArrow dtypes the default for all dtypes, then ok, maybe that'd be justified. But probably not if it's just for the sake of strings, for which I think that
1. recommending `pip install pandas[pyarrow]` in all instructions
2. auto-inferring pyarrow strings if pyarrow is installed

should be enough

Reply:
Sounds good to me.

I'll add this to the PDEP.

Reply:
> And the wheel size on PyPI is still ~40MB, so we'd be noticeably increasing the load on PyPI by making PyArrow required

Out of curiosity - is this something that the PyPi maintainers have mentioned as a problem?

If so, the plot twist is then that we should really be pushing AWS users towards using the pre-provided image, rather than building their own from scratch.  I believe that would forgo hitting PyPi altogether, and even if it doesn't, it is still smaller than what people are building themselves (see https://github.com/pandas-dev/pandas/issues/54466#issuecomment-2102794098)

Reply:
> If so, the plot twist is then that we should really be pushing AWS users towards using the pre-provided image, rather than building their own from scratch. I believe that would forgo hitting PyPi altogether, and even if it doesn't, it is still smaller than what people are building themselves (see [#54466 (comment)](https://github.com/pandas-dev/pandas/issues/54466#issuecomment-2102794098))

Wouldn't we then be asking our users to be dependent on the AWS people updating their pre-provided image whenever we created a new release of `pandas` or the arrow team did a new release of `pyarrow` ?

Reply:
Yea I think that is generally how lambda works, even with the overall Python version. It's not quite an "anything goes" type of execution environment

Reply:
There's this post from 2021, in which they write about the monthly PyPI bill being almost 2 million US dollars

https://dustingram.com/articles/2021/04/14/powering-the-python-package-index-in-2021/

Reply:
> There's this post from 2021, in which they write about the monthly PyPI bill being almost 2 million US dollars
> 
> https://dustingram.com/articles/2021/04/14/powering-the-python-package-index-in-2021/

I'm not too worried about this - PyPI's bill is subsidized anyways. Of course, it is nice to reduce our load on PyPI, but I don't think we are close to being the worse offenders here (those would probably be something like tensorflow and pytorch), and it's important to keep in mind that a lot of people have pyarrow installed for whatever reason anyways.

I would mostly only be concerned about an increase in size in our own pandas package (since PyPI does limit the total size of all packages uploaded by a project, and raising the limit is a manual and annoying process)

Reply:
I've made another go at clarifying point1, incorporating the feedback here.

PTAL @WillAyd @MarcoGorelli

Reply:
Thanks for the feedback, but I still disagree with calling the AWS-provided layer a "workaround" - essentially it is the canonical approach to solve an issue that has existed for years with pandas and lambda functions.

A quick google search for something how "how to run pandas on aws lambda" yields a slew of conflicting results on how to get this to work. If the AWS-provided layer is a workaround, then what are we calling the proper approach?

https://stackoverflow.com/questions/36054976/pandas-aws-lambda
https://stackoverflow.com/questions/53824556/how-to-install-numpy-and-pandas-for-aws-lambdas
https://medium.com/swlh/how-to-add-python-pandas-layer-to-aws-lambda-bab5ea7ced4f
https://medium.com/@johnnymao/how-to-use-pandas-in-your-aws-lambda-function-c3ce29f6f189
https://medium.com/@shimo164/lambda-layer-to-use-numpy-and-pandas-in-aws-lambda-function-8a0e040faa18
https://www.youtube.com/watch?v=1UDEp90S9h8